src/lib/helpers.ts

```diff
@@ -27,3 +27,21 @@ export function castFromString(value: string, type: DataSetColumnType) {
   }
   return value;
 }
+
+/**
+ * Takes a proposed new column name and returns a valid name that does not erase
+ * an existing column name. So if the proposed new name already exist in
+ * existing names, it appends an integer that increments until the new name
+ * becomes unique
+ *
+ * @param newName the proposed new column name
+ * @param existingNames the existing columns names
+ */
+export function newColumnName(newName: string, existingNames: string[]): string {
```

Comment:
I think a name with an action verb would be interesting, like `generateColumnName` or something like that